src/errno.rs

```diff
@@ -39,34 +52,80 @@ cfg_if! {
     }
 }
 
-/// Sets the platform-specific errno to no-error
-fn clear() {
-    // Safe because errno is a thread-local variable
-    unsafe {
-        *errno_location() = 0;
-    }
-}
-
 /// Returns the platform-specific value of errno
+#[deprecated(since = "0.28.0", note = "please use `Errno::last_raw()` instead")]
 pub fn errno() -> i32 {
-    unsafe { *errno_location() }
+    Errno::last_raw()
 }
 
 impl Errno {
+    /// Returns the current value of errno
     pub fn last() -> Self {
-        last()
+        Self::from_raw(Self::last_raw())
     }
 
-    pub fn desc(self) -> &'static str {
-        desc(self)
+    /// Returns the current raw i32 value of errno
+    pub fn last_raw() -> i32 {
+        unsafe { *errno_location() }
+    }
+
+    /// Sets the value of errno.
+    ///
+    /// # Example
+    /// ```
+    /// use nix::errno::Errno;
+    ///
+    /// Errno::set(Errno::EIO);
+    ///
+    /// assert_eq!(Errno::last(), Errno::EIO);
+    /// ```
+    pub fn set(errno: Self) {
```

Comment:
This API is redundant: the user must type "Errno" twice.  I think it would be better like this:
```rust
pub fn set(self);
```
that way a user could call it simply by:
```rust
Errno::EINVAL.set();
```

Reply:
I find this confusing, but using self does not stop one from still doing `Errno::set(err)` syntax, so I'm fine with this.
Done

Reply:
> ```rs
> Errno::EINVAL.set();
> ```
> I find this confusing,

This indeed feels a little bit weird, but anyway, it is ok, no strong view on this.